flang/lib/Lower/Bridge.cpp

```diff
@@ -1481,6 +1481,37 @@ class FirConverter : public Fortran::lower::AbstractConverter {
         curEval = &*std::next(curEval->getNestedEvaluations().begin());
       }
     }
+    // If a 'sections' construct is encountered, loop over all 'section'
+    // directives nested within and generate FIR for the same. The individual
+    // 'section' inside a 'sections' construct are not constructs rather
+    // 'Section Blocks'. Hence, nested handling of constructs doesn't apply and
+    // we need special handling
+    if (const auto &ompSections =
+            std::get_if<Fortran::parser::OpenMPSectionsConstruct>(&omp.u)) {
+      const auto &ompSectionBlocks =
+          std::get<Fortran::parser::OmpSectionBlocks>(ompSections->t);
+      auto &sectionsBlockEvaluationList = curEval->getNestedEvaluations();
+      std::list<Fortran::lower::pft::Evaluation>::iterator
+          sectionsBlockEvalIterator = sectionsBlockEvaluationList.begin();
+      for (const auto &block : ompSectionBlocks.v) {
+        auto insertPt = builder->saveInsertionPoint();
+        // create a 'section' operation for every 'Section Block'
+        genOpenMPSectionsBlock(*this, *curEval);
+        for (auto it = block.begin(); it != block.end(); it++) {
+          if (sectionsBlockEvalIterator == sectionsBlockEvaluationList.end()) {
+            break;
+          }
```

Comment:
Why is this required?
Also, as per LLVM coding guidelines we do not need the curlies here.

Reply:
Thanks for the review. There are two things here: the `EvaluationList` within a `sections` and `ompSectionBlocks`. This check is basically looking that the `EvaluationList` shouldn't exceed the execution statement list that `ompSectionBlocks` has.

Ideally, I don't think the nested evaluation list will ever be greater than the `ompSectionBlocks` list, but I thought it would be good to keep this check anyway. Should I remove it?

Reply:
If it is not needed then it will be better to remove it. But before you remove it, please have a look elsewhere whether such code is present.

Reply:
I looked elsewhere: did not find such checks. So I have removed it.